File: python/infinicore/ops/rotg.py
from infinicore.lib import _infinicore
from infinicore.tensor import Tensor


def rotg(x: Tensor, y: Tensor, c: Tensor, s: Tensor):
    _infinicore.rotg_(x._underlying, y._underlying, c._underlying, s._underlying)
    return x, y, c, s

File: src/infinicore/ops/rotg/rotg.cc
#include "infinicore/ops/rotg.hpp"
#include "../../utils.hpp"

namespace infinicore::op {

common::OpDispatcher<Rotg::schema> &Rotg::dispatcher() {
    static common::OpDispatcher<Rotg::schema> dispatcher_;
    return dispatcher_;
};

void Rotg::execute(Tensor a, Tensor b, Tensor c, Tensor s) {
    infinicore::context::setDevice(a->device());
    dispatcher().lookup(a->device().getType())(a, b, c, s);
}

void rotg_(Tensor a, Tensor b, Tensor c, Tensor s) {
    Rotg::execute(a, b, c, s);
}

} // namespace infinicore::op